packages/authentication/msal-browser/src/msalBrowserAccessTokenProvider.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  AuthenticationResult,
+  InteractionRequiredAuthError,
+  InteractionType,
+} from "@azure/msal-browser";
+import {
+  AccessTokenProvider,
+  AllowedHostsValidator,
+  validateProtocol,
+} from "@microsoft/kiota-abstractions";
+
+import { MSALBrowserAuthenticationConfig } from "./msalBrowserAuthenticationConfig";
+
+/** Access token provider that leverages the MSALBrowser Identity library to retrieve an access token. */
+export class MSALBrowserAccessTokenProvder implements AccessTokenProvider {
+  /**
+   *@param msalBrowserAuthenticationConfig The MSALBrowser public client application, scopes, account type and interaction type to use for authentication.
+   *@param allowedHosts The allowed hosts to use for authentication.
+   */
+  public constructor(
+    private msalBrowserAuthenticationConfig: MSALBrowserAuthenticationConfig,
+    allowedHosts: Set<string>
+  ) {
+    if (!msalBrowserAuthenticationConfig) {
+      throw new Error(
+        "Please pass valid PublicClientApplication instance, AccountInfo, Scopes and InteractionType"
+      );
+    }
+    this.allowedHostsValidator = new AllowedHostsValidator(allowedHosts);
+  }
+  private readonly allowedHostsValidator: AllowedHostsValidator;
+
+  /**
+   * @inheritdoc
+   */
+  public async getAuthorizationToken(url?: string): Promise<string> {
+    if (!url || !this.allowedHostsValidator.isUrlHostValid(url)) {
+      return "";
+    }
+    validateProtocol(url);
+
+    const scopes =
+      this.msalBrowserAuthenticationConfig &&
+      this.msalBrowserAuthenticationConfig.scopes;
+    const account =
+      this.msalBrowserAuthenticationConfig &&
+      this.msalBrowserAuthenticationConfig.account;
+    const error = new Error();
```

Comment:
I'd argue for duplication and instantiating it only if needed. Hopefully the vast majority of executions do not throw an error.